src/tree/hist/evaluate_splits.h

```diff
@@ -429,7 +430,7 @@ class HistEvaluator {
               all_entries[worker * entries.size() + nidx_in_set].split);
         }
       }
-      if (is_secure_) {
+      if (is_secure_ && is_col_split_) {
```

Comment:
Looking at the code, the `is_col_split_` must be true.

Reply:
ah indeed, thanks! 
removed the redundant condition.